ortools/linear_solver/unittests/xpress_interface.cc

```diff
@@ -254,7 +281,22 @@ namespace operations_research {
     EXPECT_EQ(getter.getNumVariables(), 502);
   }
   
+  TEST(XpressInterface, VariablesName) {
+    UNITTEST_INIT_MIP();
+    solver.MakeRowConstraint(-solver.infinity(), 0);
+
+    std::string pi("Pi");
+    std::string secondVar("Name");
+    MPVariable* x1 = solver.MakeNumVar(-1., 5.1, pi);
+    MPVariable* x2 = solver.MakeNumVar(3.14, 5.1, secondVar);
 
+    // Extracting row & column names requires writing an MPS first
+    std::string tmpName = std::string(std::tmpnam(nullptr)) + ".mps";
```

Comment:
he use of `tmpnam' is dangerous, better use `mkstemp'

Reply:
see [erik's answer](https://stackoverflow.com/questions/21295974/are-there-any-reasons-to-avoid-using-tmpnam-to-get-a-name-for-a-temporary-file)

Reply:
out of scope, sorry